### DISCUSSION TITLE
How to setup with existing contenteditable without adding extra markup.

### DISCUSSION THREAD

**Discussion by B-Esmaili** [ueberdosis/tiptap#5834] [General Questions] [answered]
I am using this library in a svelete project. I have a contenteditable div already and i am gonna setup the editor in a way that i only call the editor's api over the selection. the problem is the setup adds extra set of DOM elements inside my contenteditable like following : 

```
<div contenteditable class="s-66za1fzHfg4L">
   Content
   <div  translate="no" class="tiptap ProseMirror" role="textbox" aria-label="Rich-Text Editor">
      <p>
      <br class="ProseMirror-trailingBreak"></p>
   </div>
   </div>

```   
   is there a way to setup with no extra dom elements and only execute the library API over existing contenteditable?

**Comment by nperez0111** [comment, marked as answer]
No, that is not how the library is meant to be used at all. It expects full control of the contenteditable div.

**Reply by B-Esmaili** [reply]
Thanks for you response , Do you know any alternative giving me that option?

**Reply by nperez0111** [reply]
None, as far as I know, you have to give up the contenteditble element to a library.
